tests/test_tree.py

```diff
@@ -496,3 +498,89 @@ def test_X_idx_sorted(fake_data, val):
             match="The parameter 'X_idx_sorted' is deprecated and has no effect."
     ):
         tree.fit(X, y, X_idx_sorted=X_idx_sorted)
+
+
+def test_apply(veterans):
+    X, y = veterans
+    X = X.loc[:, "Karnofsky_score"].values[:, numpy.newaxis]
+    X = X.astype(numpy.float32)
+
+    tree = SurvivalTree(max_depth=2, max_features=1)
+    tree.fit(X, y)
+
+    X_trans = tree.apply(X)
+
+    assert X_trans.shape[0] == X.shape[0]
+    assert all(X_trans >= 0)
+    assert all(X_trans < tree.tree_.node_count)
+
+    X_path = tree.decision_path(X).toarray()
+
+    assert X_path.shape[0] == X.shape[0]
+    assert X_path.shape[1] == tree.tree_.node_count
+
+    ones = X_path[numpy.arange(X.shape[0]), X_trans]
+    assert_array_equal(ones, numpy.ones(X.shape[0]))
+
+
+def test_apply_sparse(veterans):
+    X, y = veterans
+    X = X.loc[:, "Karnofsky_score"].values[:, numpy.newaxis]
+    X = X.astype(numpy.float32)
+    X_sparse = sparse.csr_matrix(X)
+    tree = SurvivalTree(max_depth=2, max_features=1)
+    tree.fit(X_sparse, y)
+
+    X_trans = tree.apply(X_sparse)
+
+    assert X_trans.shape[0] == X.shape[0]
+    assert all(X_trans >= 0)
+    assert all(X_trans < tree.tree_.node_count)
+
+    X_path = tree.decision_path(X_sparse).toarray()
+
+    assert X_path.shape[0] == X.shape[0]
+    assert X_path.shape[1] == tree.tree_.node_count
+
+    ones = X_path[numpy.arange(X.shape[0]), X_trans]
+    assert_array_equal(ones, numpy.ones(X.shape[0]))
+
+
+@pytest.mark.parametrize("seed", range(100))
```

Comment:
It's good that you checked that this test passes on many seeds but I think in the long run it's better to only run on a single seed to avoid wasting CI resources.